cnd/src/http_api/herc20_hbit/bob.rs

```diff
@@ -0,0 +1,324 @@
+                        state:
+                            hbit::State::Funded {
+                                htlc_location,
+                                fund_transaction,
+                                ..
+                            },
+                        transient_redeem_identity,
+                        transient_refund_identity: transient_refund_sk,
+                        final_refund_identity,
+                        expiry,
+                        ..
+                    },
+                secret_hash,
+                ..
+            } => {
+                let network = bitcoin::Network::Regtest;
+                let spend_output = {
+                    let transient_refund_identity =
+                        identity::Bitcoin::from_secret_key(&*crate::SECP, &transient_refund_sk);
+                    let htlc = build_bitcoin_htlc(
+                        *transient_redeem_identity,
+                        transient_refund_identity,
+                        *expiry,
+                        *secret_hash,
+                    );
+
+                    let previous_output = *htlc_location;
+                    let value = bitcoin::Amount::from_sat(
+                        fund_transaction.output[htlc_location.vout as usize].value,
+                    );
+                    let input_parameters =
+                        htlc.unlock_after_timeout(&*crate::SECP, *transient_refund_sk);
+
+                    SpendOutput::new(previous_output, value, input_parameters, network)
+                };
+
+                let primed_transaction =
+                    spend_output.spend_to(final_refund_identity.clone().into());
+                let transaction = sign_with_fixed_rate(&*crate::SECP, primed_transaction)?;
+
+                Ok(BroadcastSignedTransaction {
+                    transaction,
+                    network,
+                })
+            }
+            _ => anyhow::bail!(ActionNotFound),
+        }
+    }
+}
+
+impl AlphaEvents
+    for BobSwap<asset::Erc20, asset::Bitcoin, herc20::Finalized, hbit::FinalizedAsFunder>
+{
+    fn alpha_events(&self) -> Option<LedgerEvents> {
+        match self {
+            BobSwap::Created { .. } => None,
+            BobSwap::Finalized {
+                alpha_finalized:
+                    herc20::Finalized {
+                        state: herc20_state,
+                        ..
+                    },
+                ..
+            } => Some(From::<herc20::State>::from(herc20_state.clone())),
+        }
+    }
+}
+
+impl BetaEvents
+    for BobSwap<asset::Erc20, asset::Bitcoin, herc20::Finalized, hbit::FinalizedAsFunder>
+{
+    fn beta_events(&self) -> Option<LedgerEvents> {
+        match self {
+            BobSwap::Created { .. } => None,
+            BobSwap::Finalized {
+                beta_finalized: hbit::FinalizedAsFunder { state, .. },
+                ..
+            } => Some(From::<hbit::State>::from(state.clone())),
```

Comment:
This doesn't cut it?

```suggestion
            } => Some(LedgerEvents::from(state.clone())),
```

Reply:
Even easier with `.into()`!